mock/mock.go

```diff
@@ -527,6 +527,47 @@ func (m *Mock) AssertNotCalled(t TestingT, methodName string, arguments ...inter
 	return true
 }
 
+// IsMethodCallable checking that the method can be called
+// If the method was called more than `Repeatability` return false
+func (m *Mock) IsMethodCallable(t TestingT, methodName string, arguments ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	var isCallable = false
+	for _, v := range m.ExpectedCalls {
+		if v.Method != methodName {
+			continue
+		}
+		if len(arguments) != len(v.Arguments) {
+			continue
+		}
+		if v.Repeatability < v.totalCalls {
+			continue
+		}
+		if isArgsEqual(v.Arguments, arguments) {
+			isCallable = true
+			break
+		}
+	}
+	return isCallable
+}
+
+// Compares arguments
+func isArgsEqual(expected Arguments, args []interface{}) bool {
+	if len(args) != len(args) {
+		return false
```

Comment:
This code is unreachable.  `len(args)` will always be equal to `len(args)`.

Reply:
Did you mean to test `len(expected)` vs `len(args)`?